src/lib.rs

```diff
@@ -129,9 +127,28 @@ impl Order {
         self.account.get(&mut self.nonce, challenge_url).await
     }
 
-    /// Get the current state of the order
-    pub async fn state(&mut self) -> Result<OrderState, Error> {
-        self.account.get(&mut self.nonce, &self.order_url).await
+    /// Refresh the current state of the order
+    pub async fn refresh(&mut self) -> Result<&OrderState, Error> {
```

Comment:
Not sure about this name. I also tried `update()` but I think I like `refresh()` better.